src/account.ts

```diff
@@ -393,24 +410,23 @@ export class Account {
 
     /**
      * Returns account details in terms of authorized apps and transactions
-     * @returns {Promise<any>}
+     * @returns {Promise<{ authorizedApps: AccountAuthorizedApp[] }>}
      */
-    async getAccountDetails(): Promise<any> {
+    async getAccountDetails(): Promise<{ authorizedApps: AccountAuthorizedApp[] }> {
         // TODO: update the response value to return all the different keys, not just app keys.
         // Also if we need this function, or getAccessKeys is good enough.
         const accessKeys = await this.getAccessKeys();
-        const result: any = { authorizedApps: [], transactions: [] };
-        accessKeys.map((item) => {
-            if (item.access_key.permission.FunctionCall !== undefined) {
-                const perm = item.access_key.permission.FunctionCall;
-                result.authorizedApps.push({
-                    contractId: perm.receiver_id,
-                    amount: perm.allowance,
+        const authorizedApps = accessKeys
+            .filter(item => item.access_key.permission !== 'FullAccess')
```

Comment:
Check if the permission is `FullAccess`, if not then it has to be `FunctionCallPermissionView`.